File: Java/src/dumbGame/DumbGameChecker.java
package dumbGame;

import com.sonicscholar.simpleBoardGame.GameChecker;
import com.sonicscholar.simpleBoardGame.GameEngine;
import com.sonicscholar.simpleBoardGame.Player;

public class DumbGameChecker implements GameChecker {
    GameEngine _gameEngine;
    @Override
    public GameEngine getGameEngine() {
        return _gameEngine;
    }

    @Override
    public void setGameEngine(GameEngine gameEngine) {
        _gameEngine = gameEngine;
    }

    @Override
    public boolean isGameOver() {
        var board = _gameEngine.getBoard();
        for (int row = 0; row < board.getHeight(); row++) {
            for (int col = 0; col < board.getWidth(); col++) {
                if (board.isEmpty(row, col))
                    return false;
            }
        }

        return true;
    }

    @Override
    public Player getWinner() {
        return null;
    }
}

File: C_Sharp/SonicScholar.SimpleBoardGame.Framework/DumbGame/DumbGameChecker.cs
using SonicScholar.SimpleBoardGame.Framework;

namespace DumbGame
{
    public class DumbGameChecker : IGameChecker
    {
        public IGameEngine GameEngine { get; set; }

        public bool IsGameOver
        {
            get
            {
                var board = GameEngine.Board;
                for (int row = 0; row < board.Height; row++)
                {
                    for (int col = 0; col < board.Width; col++)
                    {
                        if (board.IsEmpty(row, col))
                            return false;
                    }
                }

                return true;
            }
        }

        public IPlayer WinningPlayer => null;
    }
}